3711/1/top_gems.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'scanner/shell'
+require_relative 'scanner/yaml'
+require_relative 'parser/ruby_gems'
+require_relative 'parser/git_hub'
+require_relative 'rating'
+require_relative 'printer'
+
+# Scan input atributes from shell
+rating_args = Scanner::Shell.new(ARGV).scan
+
+# Get gem list from local file by input attributes
+gem_names = Scanner::Yaml.new(rating_args).scan
+puts "Can't find any gems" if gem_names.size.zero?
+
+unless gem_names.size.zero?
```

Comment:
I would say this logic should be processed in the Parser::RubyGems class. This class should process all arguments